test/e2e/test-postcreatehook-retry.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SRC_DIR="$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)"
+source "${SRC_DIR}/setup-shell.sh"
+
+set -x # echo commands for better debugging
+set -e # exit on error
+
+:
+: -------------------------------------------------------------------------
+: Clean up any existing test resources
+:
+echo "Cleaning up any existing resources..."
+kubectl delete controlplane cp-missing-hook --ignore-not-found=true
+kubectl delete postcreatehook missing-hook --ignore-not-found=true
+
+:
+: -------------------------------------------------------------------------
+: Test case: ControlPlane with missing PostCreateHook should not fail
+: but should retry until the hook is available
+:
+echo "Creating ControlPlane referencing a missing PostCreateHook..."
+kubectl apply -f - <<EOF
+apiVersion: tenancy.kflex.kubestellar.org/v1alpha1
+kind: ControlPlane
+metadata:
+  name: cp-missing-hook
+spec:
+  backend: shared
+  postCreateHook: missing-hook
+  waitForPostCreateHooks: true
+  type: k8s
+EOF
+
+:
+: -------------------------------------------------------------------------
+: Verify ControlPlane is not marked as failed while waiting for hook
+:
+echo "Waiting 10s to check that ControlPlane is not marked as failed..."
+sleep 10
+
+echo "ControlPlane status after 10s (should NOT be failed):"
```

Comment:
Being a test, shouldn't this fail if the `kubectl get` returns the unexpected result?